src/Compilers/Core/Portable/InternalUtilities/ImmutableArrayExtensions.cs

```diff
@@ -9,24 +9,7 @@ namespace Roslyn.Utilities
     internal static class ImmutableArrayExtensions
     {
         internal static ImmutableArray<T> ToImmutableArrayOrEmpty<T>(this IEnumerable<T> items)
-        {
-            if (items == null)
-            {
-                return ImmutableArray.Create<T>();
-            }
-
-            return ImmutableArray.CreateRange<T>(items);
-        }
-
-        internal static ImmutableArray<T> ToImmutableArrayOrEmpty<T>(this ImmutableArray<T> items)
-        {
-            if (items.IsDefault)
-            {
-                return ImmutableArray.Create<T>();
-            }
-
-            return items;
-        }
+            => items == null ? ImmutableArray<T>.Empty : ImmutableArray.CreateRange(items);
```

Comment:
I would suggest this goes into master for perf (in case there wasn't an overload for zero-args and we were allocating a params array), but turns out that [`ImmutableArray.Create()`](https://github.com/dotnet/corefx/blob/master/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs#L30) calls through to `ImmutableArray.Empty`, so this is fine in this branch.

Also to note (to other reviewers), `ImmutableArray.CreateRange` does a dynamic check to see if the parameter [is a boxed ImmutableArray](https://github.com/dotnet/corefx/blob/master/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs#L102), so there shouldn't be any perf issues from overloads suddenly calling `CreateRange` instead of passing back the same array in the other overload.